ckanapi/remoteckan.py

```diff
@@ -82,12 +99,25 @@ def call_action(self, action, data_dict=None, context=None, apikey=None,
         if self.get_only:
             status, response = self._request_fn_get(url, data_dict, headers, requests_kwargs)
         else:
-            status, response = self._request_fn(url, data, headers, files, requests_kwargs)
+            status, response = self._request_fn(url, data, headers, files,
+                                                progress, requests_kwargs)
         return reverse_apicontroller_action(url, status, response)
 
-    def _request_fn(self, url, data, headers, files, requests_kwargs):
-        r = self.session.post(url, data=data, headers=headers, files=files,
-            allow_redirects=False, **requests_kwargs)
+    def _request_fn(self, url, data, headers, files, progress, requests_kwargs):
+        # use streaming
+        if files:
+            files['upload'] = (getattr(files['upload'], 'name', 'upload'),
+                               files['upload'])
+            data.update(files)
+            m = MultipartEncoder(data)
+            if progress:
+                m = MultipartEncoderMonitor(m, self._mkcallback(m))
+            headers.update({'Content-Type': m.content_type})
+            r = self.session.post(url, data=m, headers=headers,
+                                  allow_redirects=False, **requests_kwargs)
+        else:
+            r = self.session.post(url, data=data, headers=headers, files=files,
+                    allow_redirects=False, **requests_kwargs)
```

Comment:
baking a particular progress bar implementation into RemoteCKAN isn't cool. Move this to ckanapi.cli and expose requests callback parameter so users of the library can implement any interface they like instead.